src/stylesheets/components/_icon-list.scss

```diff
@@ -0,0 +1,152 @@
+// Variables
+
+// Size icons based on the theme-type-scale-[size] number and this factor
+$icon-list-icon-size-factor: 1.5;
+// Center the icon to the title text, works for all size variants
+$icon-list-icon-margin-top: -1.5%;
+// Fine tune the space between the icon and content with this factor
+$icon-list-icon-padding-left-factor: 0.4;
+// Fine tune the title top padding given a line height of 2
+$icon-list-title-padding-top: 0.2em;
+
+$icon-list-font-sizes: (
+  "xs" $theme-type-scale-xs,
+  "sm" $theme-type-scale-sm,
+  "md" $theme-type-scale-md,
+  "lg" $theme-type-scale-lg,
+  "xl" $theme-type-scale-xl,
+  "2xl" $theme-type-scale-2xl,
+  "3xl" $theme-type-scale-3xl
+);
+
+$type-scale: font-size($theme-icon-list-font-family, $theme-body-font-size);
+
+@include override-prose {
+  .usa-icon-list {
+    @include typeset($theme-icon-list-font-family);
+    @include unstyled-list;
+    @include u-measure(5);
+  }
+
+  .usa-icon-list__item {
+    display: flex;
+    font-size: size($theme-icon-list-font-family, $theme-icon-list-font-size);
+    position: relative;
+
+    & + .usa-icon-list__item {
+      padding-top: units(1.5);
+    }
+  }
+}
+
+// Allow setting icon color on the icon-list element
+.usa-icon-list {
+  @each $color, $grades in $all-project-colors {
+    @each $grade, $value in $grades {
+      @if $value {
+        $prefix: if($grade != "default", #{$color}-#{$grade}, #{$color});
+        &--#{$prefix} {
+          .usa-icon-list__icon {
+            color: color($prefix);
+          }
+        }
+      }
+    }
+  }
+}
+
+.usa-icon-list__icon {
+  .usa-icon {
+    // Set the height and width of the icon based on the size variant and factor
+    display: block;
+    height: $type-scale * $icon-list-icon-size-factor;
+    margin-top: $icon-list-icon-margin-top;
+    position: relative;
+    width: $type-scale * $icon-list-icon-size-factor;
+  }
+}
+
+// Generate responsive variants
+// Create a map for looping that includes a default or no-breakpoint version
+$this-null: (
+  "none": null,
+);
+$icon-list-breakpoints: map-merge($this-null, $system-breakpoints);
+@each $mq-key, $mq-value in $icon-list-breakpoints {
+  @each $token, $val in $icon-list-font-sizes {
+    // Set the prefix
+    // An empty string if "none"
+    $prefix: false;
+    @if $mq-key == "none" {
+      $prefix: "";
+    }
+    // Or the standard prefix if the breakpoint is output
+    @else if map-get($theme-utility-breakpoints, $mq-key) {
+      $prefix: "#{$mq-key}\\:";
+    }
+    @if $prefix {
+      $this-type-scale: font-size($theme-icon-list-font-family, $token);
+      @include at-media($mq-key) {
+        .#{$prefix}usa-icon-list--size-#{$token} {
```

Comment:
Could we simplify this with an attribute selector like `.usa-icon-list[class*="usa-icon-list--size-"]`?

Output CSS
![image](https://user-images.githubusercontent.com/3385219/109835919-4474fe80-7c09-11eb-9cd6-f760b40f7129.png)

Versus current
![image](https://user-images.githubusercontent.com/3385219/109836013-5eaedc80-7c09-11eb-9b51-4518e86aafc1.png)

Reply:
Good idea